File: VibesTCGScoreTracker/VibesTCGScoreTracker/Models/GameAction.swift
//
//  GameAction.swift
//  VibesTCGScoreTracker
//
//  Tracks actions for history log and undo functionality
//

import Foundation

/// Represents an action taken during the game (for undo/history)
struct GameAction: Identifiable {
    let id = UUID()
    let timestamp = Date()
    let description: String
    let undoAction: () -> Void

    /// Format timestamp for display
    var timeString: String {
        let formatter = DateFormatter()
        formatter.timeStyle = .short
        return formatter.string(from: timestamp)
    }
}

File: VibesTCGScoreTrackerAndroid/app/src/main/java/com/vibestcg/scoretracker/data/GameAction.kt
package com.vibestcg.scoretracker.data

import java.text.SimpleDateFormat
import java.util.*

/**
 * Represents an action taken during the game (for undo/history)
 */
data class GameAction(
    val id: String = UUID.randomUUID().toString(),
    val timestamp: Long = System.currentTimeMillis(),
    val description: String,
    val undoAction: () -> Unit
) {
    /**
     * Format timestamp for display
     */
    val timeString: String
        get() {
            val sdf = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
            return sdf.format(Date(timestamp))
        }
}
